### PR TITLE
NOX-426 swagger language parameter added 

### PR DESCRIPTION
**Pull request by msadikkoseaupa**

### CODE REVIEW

File: src/Nox.Lib/Presentation/Api/OData/ApiRouteQueryStringOperationFilter.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+public class ApiRouteQueryStringOperationFilter: IOperationFilter
+{
+    private readonly IEnumerable<string> _auditEntityPaths;
+
+    public ApiRouteQueryStringOperationFilter(NoxSolution noxSolution)
+    {
+        _auditEntityPaths = 
+            noxSolution.Domain!.Entities.Where(e => e.Persistence.IsAudited)
+                .Select(e=> string.Concat(noxSolution.Presentation.ApiConfiguration.ApiRoutePrefix.Trim('/'), "/", e.PluralName));
```
Comment (rochar) [inline src/Nox.Lib/Presentation/Api/OData/ApiRouteQueryStringOperationFilter.cs]: ApiRoute Prefix is already sanitize. Maybe you can also consider using string interpolation
```suggestion
                .Select(e=> string.Concat(noxSolution.Presentation.ApiConfiguration.ApiRoutePrefix, "/", e.PluralName));
```

File: src/Nox.Lib/Presentation/Api/OData/ApiRouteQueryStringOperationFilter.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    }
+    
+    public void Apply(OpenApiOperation operation, OperationFilterContext context)
+    {
+        if (context.ApiDescription.HttpMethod == HttpMethod.Get.Method && _auditEntityPaths.Any(p => context.ApiDescription.RelativePath!.Contains(p)))
```
Comment (rochar) [inline src/Nox.Lib/Presentation/Api/OData/ApiRouteQueryStringOperationFilter.cs]: _auditEntityPaths is a IEnumerable with a Linq so it always going to execute the query every time is called. Consider Hashset with all values computed once

File: src/Nox.Lib/Presentation/Api/OData/ApiRouteQueryStringOperationFilter.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        {
+            operation.Parameters.Add(new OpenApiParameter
+            {
+                Name = "lang",
```
Comment (rochar) [inline src/Nox.Lib/Presentation/Api/OData/ApiRouteQueryStringOperationFilter.cs]:
```suggestion
                Name = QueryParams.Language,
```

File: src/Nox.Lib/Presentation/Api/OData/ApiRouteQueryStringOperationFilter.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+                Description = "language parameter",
+                In = ParameterLocation.Query,
+                Style = ParameterStyle.Simple,
+                Schema = new OpenApiSchema
+                {
+                    Type = "string",
+                    Pattern = "^[a-z]{2}-[A-Z]{2}$"
```
Comment (rochar) [inline src/Nox.Lib/Presentation/Api/OData/ApiRouteQueryStringOperationFilter.cs]: see the Nox.Type culture code regular expressions. And in Nox.Solution.Localizaton you can see a unique regular expression that merges all of them

**Review by rochar — CHANGES_REQUESTED** [review]

File: src/Nox.Lib/Presentation/Api/OData/ApiRouteQueryStringOperationFilter.cs (hunk 1 above)
Comment (msadikkoseaupa) [reply src/Nox.Lib/Presentation/Api/OData/ApiRouteQueryStringOperationFilter.cs]: Desired changes have been applied

File: src/Nox.Lib/Presentation/Api/OData/ApiRouteQueryStringOperationFilter.cs (hunk 2 above)
Comment (msadikkoseaupa) [reply src/Nox.Lib/Presentation/Api/OData/ApiRouteQueryStringOperationFilter.cs]: Desired changes have been applied

File: src/Nox.Lib/Presentation/Api/OData/ApiRouteQueryStringOperationFilter.cs (hunk 3 above)
Comment (msadikkoseaupa) [reply src/Nox.Lib/Presentation/Api/OData/ApiRouteQueryStringOperationFilter.cs]: Desired changes have been applied

File: src/Nox.Lib/Presentation/Api/OData/ApiRouteQueryStringOperationFilter.cs (hunk 4 above)
Comment (msadikkoseaupa) [reply src/Nox.Lib/Presentation/Api/OData/ApiRouteQueryStringOperationFilter.cs]: Desired changes have been applied

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

**Review by rochar — APPROVED** [review]

**Pull request merged**